funcx_endpoint/funcx_endpoint/cli.py

```diff
@@ -315,9 +317,21 @@ def _do_start_endpoint(
     log_to_console: bool,
     no_color: bool,
 ):
+    reg_info = {}
+    if sys.stdin and not (sys.stdin.closed or sys.stdin.isatty()):
+        try:
+            stdin_data = json.loads(sys.stdin.read())
+            if not isinstance(stdin_data, dict):
+                type_name = stdin_data.__class__.__name__
+                raise ValueError(f"Expecting dictionary; got {type_name}")
```

Comment:
'dictionary' is a little vague.  Presumably if a user uses this in a script he knows it's part of start_endpoint, but it is more helpful if the error has more context.  Something along the lines of "Expecting JSON dictionary for endpoint registration info input, got {type_name} instead"?

In line 330 below which is hit whenever there's a JSON parser error (the more common alert?), it would say something like 'Invalid registration info on stdin -- (JSONDecodeError) Expecting value: line 1 column 9 (char 8)' which should mention JSON.

Reply:
Good suggestion.  I expect next-to-no users to ever see this, but a developer just might when playing around.  I'll update the text.